### PR TITLE
docs/vmanomaly: add self-monitoring section

### PR DESCRIPTION
**Pull request by fred-navruzov**
### Describe Your Changes

- Added self-monitoring guide for `vmanomaly`.
- Added cross-referencing on other pages.
- Slight improvements in wording on related pages
- Update references to v1.18.4
- [x] publish Grafana dashboard to https://grafana.com/orgs/victoriametrics/dashboards:
  https://grafana.com/grafana/dashboards/22337-victoriametrics-vmanomaly/

@AndrewChubatiuk , JFYI if it somehow impacts your work on supporting `vmanomaly` in operator.

### Checklist

The following checks are **mandatory**:

- [x] My change adheres [VictoriaMetrics contributing guidelines](https://docs.victoriametrics.com/contributing/).

### CODE REVIEW

**Review by tiny-pangolin — CHANGES_REQUESTED** [review]
Please add link before merging

File: docs/anomaly-detection/Self-monitoring.md [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+
+> **Note**: For the detailed overview of self-monitoring metrics that are produced by `vmanomaly` and how to enable their tracking for push/pull models, please refer to [monitoring](https://docs.victoriametrics.com/anomaly-detection/components/monitoring) section docs.
+
+The self-monitoring assets of `vmanomaly` include Grafana dashboard and accompanying alerting rules.
+
+## Grafana Dashboard
+
+### Overview
+
+To visualize and interact with the self-monitoring metrics, `vmanomaly` provides a [Grafana Dashboard](#TODO-add-link). It offers an overview of service health (per job, instance) and performance metrics from different operational stages, including model behavior, reader, and writer components.
```
[the remark this answers is not in the document]
Comment (fred-navruzov) [reply docs/anomaly-detection/Self-monitoring.md:29]: correct, that's why this PR is still in DRAFT and I added TODO: for adding this link (right now I'm blocked to add it)

File: docs/anomaly-detection/README.md
[the remark this answers is not in the document]
Comment (fred-navruzov) [reply docs/anomaly-detection/README.md:1]: I couldn't make a better choice other than referencing our 1st article in a series (which itself has all the cross-references to other posts in series ToC), as long as I have to use 2 different tags to isolate such series (i.e. `vmanomaly` + `product updates`). Right now I know how to only reference 1 at a time, like https://victoriametrics.com/blog/categories/product-news/ or https://victoriametrics.com/blog/categories/anomaly-detection/ but not both

The only workaround I thought of is to create new (redundant) tag (i.e. `vmanomaly-updates`) and use it for reference, but that doesn't look good to me

[the remark this answers is not in the document]
Comment (tiny-pangolin) [reply docs/anomaly-detection/README.md:1]: I think linking to https://victoriametrics.com/blog/categories/anomaly-detection/ would be a better idea in this case and update some of the tags on previous posts.

File: docs/anomaly-detection/Self-monitoring.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (tiny-pangolin) [reply docs/anomaly-detection/Self-monitoring.md]: Thanks for fixing the link I think this good to merge for now There are a couple of bugs with the dashboard I imported from Grafana. I will update you in a different thread

File: docs/anomaly-detection/README.md
[the remark this answers is not in the document]
Comment (fred-navruzov) [reply docs/anomaly-detection/README.md:1]: thanks, I'll change to https://victoriametrics.com/blog/tags/anomaly-detection/ **tag**, as long as we don't have a **category** (as we have for `product-news`)

**Review by tiny-pangolin — APPROVED** [review]
looks good we can work on getting the vmanomaly tag added to the blog and updating docs at a later time.

**Pull request merged**